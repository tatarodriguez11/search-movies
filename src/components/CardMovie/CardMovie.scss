.movie {
  text-align: center;
  background-color: var(--form-text);
  padding: 15px 20px;
  border-radius: 10px;
  width: 220px;

  .movie__title,
  .movie__year {
    margin: 0;
    color: black;
  }

  .movie__image {
    border-radius: 8px;
    margin-top: 16px;
  }
}
