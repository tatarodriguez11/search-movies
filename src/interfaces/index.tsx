export interface MovieDetailResponse {
  Title: string,
  Plot: string,
  Actors: string,
  Year: string,
  Country: string,
  Language: string,
  Genre: string,
  Director: string,
  Poster: string,
  imdbID: string
}

export interface movieService {
  search: string
}

export interface Film {
  id: string,
  title: string,
  year: string,
  image: string,
}