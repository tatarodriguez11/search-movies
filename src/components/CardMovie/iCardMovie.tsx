import { Film } from "../../interfaces";

export interface CardMovieProps {
  film: Film
}