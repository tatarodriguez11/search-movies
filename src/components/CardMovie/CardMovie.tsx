import React from 'react'
import { Link } from "react-router-dom";
import { CardMovieProps } from "./iCardMovie";
import './CardMovie.scss'

const CardMovie = ({film}: CardMovieProps) => {
  return(
    <li className="movie" key={film.id}>
      <Link to={`/movies/${film.id}`}>
        <h3 className='movie__title'>{film.title}</h3>
        <p className='movie__year'>{film.year}</p>
        <img src={film.image} alt={film.title} className='movie__image'/>
      </Link>
    </li>
  )
}

export default CardMovie