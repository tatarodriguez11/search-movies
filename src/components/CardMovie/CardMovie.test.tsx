import React from "react";
import { render } from "@testing-library/react";
import CardMovie from "./CardMovie";
import { BrowserRouter } from "react-router-dom";
import { Film } from "../../interfaces";

describe("CardMovie", () => {
  const film: Film = {
    id: "1",
    title: "A Movie",
    year: "2022",
    image: "https://example.com/movie.jpg",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <CardMovie film={film} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders movie title", () => {
    const { getByText } = render(
      <BrowserRouter>
        <CardMovie film={film} />
      </BrowserRouter>
    );
    expect(getByText(film.title)).toBeInTheDocument();
  });

  it("renders movie year", () => {
    const { getByText } = render(
      <BrowserRouter>
        <CardMovie film={film} />
      </BrowserRouter>
    );
    expect(getByText(film.year.toString())).toBeInTheDocument();
  });

  it("renders movie image", () => {
    const { getByAltText } = render(
      <BrowserRouter>
        <CardMovie film={film} />
      </BrowserRouter>
    );
    expect(getByAltText(film.title)).toBeInTheDocument();
    expect(getByAltText(film.title)).toHaveAttribute("src", film.image);
  });

  it("navigates to movie page on link click", () => {
    const { container } = render(
      <BrowserRouter>
        <CardMovie film={film} />
      </BrowserRouter>
    );
    const link = container.querySelector("a");
    expect(link).toHaveAttribute("href", `/movies/${film.id}`);
  });
});
