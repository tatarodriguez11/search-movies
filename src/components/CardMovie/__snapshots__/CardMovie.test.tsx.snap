// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMovie renders correctly 1`] = `
<DocumentFragment>
  <li
    class="movie"
  >
    <a
      href="/movies/1"
    >
      <h3
        class="movie__title"
      >
        A Movie
      </h3>
      <p
        class="movie__year"
      >
        2022
      </p>
      <img
        alt="A Movie"
        class="movie__image"
        src="https://example.com/movie.jpg"
      />
    </a>
  </li>
</DocumentFragment>
`;
